### DISCUSSION TITLE
Prisma creating unique indexes on FK columns

### DISCUSSION THREAD

**Discussion by dariusj18** [prisma/prisma#8846] [Q&A] [answered]
Given the following schema
```prisma
model User {
  id        String   @id @default(uuid()) @db.Uuid
  username  String   @unique

  posts Post[]
}

model Post {
  id         String  @id @default(uuid()) @db.Uuid

  owner      User    @relation(fields: [userId], references: [id])
  userId     String  @db.Uuid

  postType   Tag?    @relation(name: "PostPostType", fields: [postTypeId], references: [id])
  postTypeId String? @db.Uuid

  category   Tag?    @relation(name: "PostCategory", fields: [categoryId], references: [id])
  categoryId String? @db.Uuid
}

model Tag {
  id           String    @id @default(uuid()) @db.Uuid
  name         String
  tagType      TagType

  user         User      @relation(fields: [userId], references: [id])
  userId       String    @db.Uuid

  postPostType Post? @relation("PostPostType")
  postCategory Post? @relation("PostCategory")

  @@unique([name, userId, tagType])
}

enum TagType {
  POSTTYPE
  CATEGORY
}
```

why would Prisma be creating a unique index on the `Posts`'s `postTypeId` and `categoryId` columns in postgresql?

**Comment by dariusj18** [comment, marked as answer]
Looking at it again, It must be because the Tag's Post properties are not arrays.

**Reply by ryands17** [reply]
Yes exactly. 1-1 relations means that the foreign key is unique. If you don't want unique indexes to be created, use a 1-many relation (`[]`).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
